api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirEncounterServiceImpl.java

```diff
@@ -169,6 +175,24 @@ public IBundleProvider searchForEncounters(DateRangeParam date, ReferenceAndList
 		return encounterBundle;
 	}
 	
+	/**
+	 * @return true if the given tokenAndListParam contains the valueToCheck or the given
+	 *         tokenAndListParam does not restrict on the encounter-tag system
+	 */
+	protected boolean shouldIncludeResourceBasedOnEncounterTag(TokenAndListParam tokenAndListParam, String valueToCheck) {
+		final Set<String> tagsToInclude = new HashSet<>();
+		if (tokenAndListParam != null && tokenAndListParam.getValuesAsQueryTokens() != null) {
+			tokenAndListParam.getValuesAsQueryTokens().forEach(tokenOrListParam -> {
+				tokenOrListParam.getValuesAsQueryTokens().forEach(tokenParam -> {
+					if (tokenParam.getSystem().equals(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG)) {
+						tagsToInclude.add(tokenParam.getValue());
+					}
+				});
+			});
+		}
+		return tagsToInclude.isEmpty() || tagsToInclude.contains(valueToCheck);
```

Comment:
Per above, if any tags are found for the OPENMRS_FHIR_EXT_ENCOUNTER_TAG system, then restrict based on that, otherwise do not.